tests/integration_tests/modules/test_combined.py

```diff
@@ -323,15 +322,12 @@ def test_combined_cloud_config_json(
             "/run/cloud-init/combined-cloud-config.json"
         )
         data = json.loads(combined_json)
-        expected_features = get_features()
-        is_devel_pkg = (
-            "bddeb" in client.execute("cloud-init -v").split()[-1]
-            or client.settings.CLOUD_INIT_SOURCE == "IN_PLACE"
+        expected_features = json.loads(
```

Comment:
What we lose with this is visibility to whether our daily images actually did correctly disable NETPLAN_CONFIG_ROOT_ROOT_ONLY. We are now just comparing that our python features.get_features() module equals /run/cloud-init/combined-cloud-config.json regardless of whether that content has the correct published expected settings and/or is well-formed. I'm ok with not checking the actual values. But, we risk accidentally forgetting to disable some feature in the future when tip of main enables something that we want to disable on stable downstreams.

Reply:
well we do assert well-formed because you use json.loads. But, we aren't asserting anywhere the expected packaged values of our features in stable releases AFAICT.

Reply:
> The test now becomes just validation that the SUT features.get_featres() == /run/cloud-init/combined-instance-data.json["features"]

Isn't that the point of the test? We want to ensure that `cloud-init query` matches what's actually set for the features. At least that seemed to be the point when you added it. The only way to really test that the values are what we think they should be is to keep a static map of cloudinit version ranges for each series and each expected value. The problem with that is we have the same problem as before...we have to remember to update the map like we have to remember to create the quilt patches. So I don't see a ton of value to that kind of test either.

Reply:
Yes. that was the original point of the initial test when I added it- assert the whatever features were configured were exposed in /run/cloud-init/combined-instance-data.json. 

I just thought when Alberto added supplemental checks it would be nice to have that level of validation that we did package what we thought we packaged. But, that's really something that we should be checking already during both our downstream SRU reviews and validation and our quilt patch reviews. So, maybe this is not really something we need in ongoing integration tests due to the fact that we are duplicating tracking of what is already tracked in debian/patches anyway.

+1 on leaving this test flexible and limiting it to ensure the original intent that features are surfaced in the /run file. If we come up with an easy to maintain strict validation of known feature settings on each stable release we can deal with that in a separate test. I agree that it's the same problem. if we are forgetting to disable a feature via a quilt patch, we are also going to forget to update the test map.